olp-cpp-sdk-authentication/src/AuthenticationClientImpl.cpp

```diff
@@ -472,6 +472,17 @@ client::CancellationToken AuthenticationClientImpl::HandleUserRequest(
     std::time_t timestamp =
         std::chrono::system_clock::to_time_t(std::chrono::system_clock::now());
 
+    if (!settings_.use_system_time) {
```

Comment:
But what about the issue that all retries use the same timestamp which will result in failed response?

Reply:
As agreed lets increment the original sever timestamp after each retry by the number of seconds elapsed.